plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/Verifier.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.viper
+
+import org.jetbrains.kotlin.formver.viper.ast.Program
+import viper.silicon.Config
+import viper.silicon.logger.MemberSymbExLogger
+import viper.silicon.logger.`NoopSymbExLog$`
+import viper.silicon.logger.SymbExLogger
+import viper.silicon.verifier.DefaultMainVerifier
+import viper.silver.cfg.silver.SilverCfg
+import viper.silver.reporter.StdIOReporter
+
+sealed interface VerifierError {
+    val msg: String
+}
+
+data class ConsistencyError(override val msg: String) : VerifierError
```

Comment:
I see three options:
1. (Current:) Pass the string as formatted by Viper and display it as-is (which includes the position infromation).
2. Pass the string as formatted by Viper, plus all associated data, but only display the string.
3. Pass the error string and the position information and put them together into a single string on our end.

I don't really see a benefit to options 2 or 3: if we want to expand on our error reporting, it seems like doing that once we're tackling that problem and know what data we need would be easier.  Or is there an option I'm missing here?

Reply:
What I meant is deferring turning the error into a string (and thus loosing some structured information) until the end. So that when passing an `onFailure` callback you still have the extra information available for better reporting. So instead of calling `toString` on the errors already in `verify` keep the information (maybe by just wrapping the Viper error) and defer turning it into a string to the `onFailure` callback.

Reply:
Ah, sorry, I see; done.